openstack/resource_openstack_networking_router_v2.go

```diff
@@ -432,6 +446,12 @@ func resourceNetworkingRouterV2Update(ctx context.Context, d *schema.ResourceDat
 		updateGatewaySettings = true
 	}
 
+	if d.HasChange("external_qos_policy_id") {
+		updateGatewaySettings = true
+		externalQoSPolicyID := d.Get("external_qos_policy_id").(string)
+		gatewayInfo.QoSPolicyID = externalQoSPolicyID
```

Comment:
I think it's enough just to assign a value directly to the `QoSPolicyID`
```suggestion
		gatewayInfo.QoSPolicyID = d.Get("external_qos_policy_id").(string)
```